templates/dataset_inplace.cpp.in

```diff
@@ -26,7 +26,7 @@ DataArray @NAME@(DataArray &&lhs, const Variable &rhs) {
 DataArray @NAME@(DataArray &&lhs, const DataArray &rhs) {
   expect::coordsAreSuperset(lhs, rhs);
   union_or_in_place(lhs.masks(), rhs.masks());
-  return std::move(@NAME@(lhs, rhs.data()));
+  return @NAME@(std::move(lhs), rhs.data());
```

Comment:
Hmm, but won't this cause inifinte recursion now, rather than calling the other overload?

Reply:
No, the rhs has a different type. `Variable` vs `DataArray`